src/core/internal/dassert.d

```diff
@@ -60,7 +60,7 @@ template _d_assert_fail(A...)
 {
     string _d_assert_fail(B...)(
         const scope string comp, auto ref const scope A a, auto ref const scope B b)
-    if (B.length > 0)
+    if (B.length > 0 || !A.length)
```

Comment:
Doesn't that mean `assert(NonEmptyTuple == EmptyTuple)` will not work ?

Reply:
Fixed. But is this even possible to compare tuples with different lengths?

Reply:
> onlineapp.d(6): Error: mismatched tuple lengths, 3 and 1

Guess not :D